### ISSUE TITLE
Hive connector does not preserve tailing space in column name

### ISSUE THREAD

**Issue by findepi** [prestosql/presto#3461]
```
CREATE TABLE t("a " varchar);

SELECT "a " FROM t
```
-- column not found

**Comment by ebyhr** [comment]
`trimResults` in HiveUtil is the cause as far as I tested.
https://github.com/prestosql/presto/blob/dc59397753dc569b49ee8940824929502b189821/presto-hive/src/main/java/io/prestosql/plugin/hive/util/HiveUtil.java#L193

**Comment by dain** [comment]
Is this something that Hive supports?  If does it support leading spaces also?

Since Hive is using a comma separated list for column names, and I believe it has no support for escaping, my guess is they never intended to support leading/trailing spaces.

**Comment by ebyhr** [comment]
> Is this something that Hive supports? If does it support leading spaces also?

Yes, I think so if "support" means "valid" regardless of their intent. 

The following are column names @findepi added tests recently. 

https://github.com/prestosql/presto/blob/7fa92370bed82e05914f746325be41cce5decfd0/presto-testing/src/main/java/io/prestosql/testing/AbstractTestDistributedQueries.java#L1294-L1317

**Comment by findepi** [comment]
As noted in https://github.com/prestosql/presto/pull/3510#issuecomment-618926807
Hive seems not to support this fully.
We should reject leading/trailing space and commas (https://github.com/prestosql/presto/issues/3537) as impossible names.

**Comment by ankitdixit** [comment]
https://github.com/prestosql/presto/pull/4666
